romancal/skycell/match.py

```diff
@@ -0,0 +1,232 @@
+        """circumference of the rectangular footprint"""
+        return sum(
+            sga.length(
+                self.vectorpoint_corners[index], self.vectorpoint_corners[index + 1]
+            )
+            for index in range(-1, len(self.vectorpoint_corners) - 1)
+        )
+
+    @cached_property
+    def polygon(self) -> sgp.SingleSphericalPolygon:
+        """spherical polygon representing this image footprint"""
+        return sgp.SingleSphericalPolygon(
+            points=self.vectorpoint_corners,
+            inside=self.vectorpoint_center,
+        )
+
+    @cached_property
+    def area(self) -> float:
+        """area of this footprint on the sphere in degrees squared"""
+        return self.polygon.area()
+
+    @cached_property
+    def possibly_intersecting_projregions(self) -> int:
+        """number of possibly intersecting projection regions"""
+        if self.area > sc.SkyCell.area:
+            return (
+                # the number of times the smallest projection region could fit in the image footprint
+                np.ceil(self.area / sc.ProjectionRegion.MIN_AREA)
+                # plus multiplier for partial intersections on the perimeter
+                * 8
+            )
+        else:
+            # 4 foundational intersections
+            return 4
+
+    @cached_property
+    def possibly_intersecting_skycells(self) -> int:
+        """number of possibly intersecting skycells"""
+        if self.polygon.area() > sc.SkyCell.area:
+            return (
+                # number of times a skycell could fit in the image footprint
+                np.ceil(self.polygon.area() / sc.SkyCell.area)
+                # plus multiplier for partial intersections on the perimeter
+                * 8
+            )
+        else:
+            # 4 foundational intersections
+            return 4
+
+
+def find_skycell_matches(
+    image_corners: list[tuple[float, float]] | NDArray[float] | WCS,
+    image_shape: tuple[int, int] | None = None,
+    skymap: sc.SkyMap = None,
+) -> tuple[list[int], list[int]]:
+    """
+    Find sky cells overlapping the provided image footprint
+
+    Parameters
+    ----------
+    image_corners : Either a squence of 4 (ra, dec) pairs, or
+        equivalent 2-d numpy array, or
+        a GWCS instance. The instance must have either the bounding_box or
+        pixel_shape attribute defined, or the following image_shape argument
+        must be supplied
+    image_shape : image shape to be used if a GWCS instance is supplied
+        and does not contain a value for either the bounding_box or
+        pixel_shape attributes. Default value is None.
+    skymap: SkyMap
+        sky map instance (defaults to global SKYMAP)
+
+    Returns
+    -------
+    A sequence of the indices of all projection regions that overlap the supplied image
+    (in the referenced projection region table). The indices may be used to obtain all
+    necessary information about the projection regions.
```

Comment:
Is this description of the return value right?  It looks to me like it's the skycell indices rather than the projection regions, and it's a tuple with two elements, the first of which I think we want, the second of which it's less clear to me is what we want?